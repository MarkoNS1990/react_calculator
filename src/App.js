function App() {
  return (
    <div className="App">
      <h1>Calculator with React.</h1>

    </div>
  );
}

export default App;
