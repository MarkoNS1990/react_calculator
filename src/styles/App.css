.container {
  width: 700px;
  border: 1px solid black;
  margin: auto;
  display: flex;
  flex-direction: column;
}

.my-button {
  padding: 10px;
  width: 25%;
  font-size: 2rem;
  border: 1px solid black;
}

.zero {
  width: 50%;
  font-size: 2rem;
  border: 1px solid black;
}

.my-button:hover,
.zero:hover {
  cursor: pointer;
}

.result {
  height: 100px;
  background-color: gray;
  font-weight: bold;
  text-align: right;
  color: #fff;
  padding: 10px;
  font-size: 5rem;
  margin: 0;
  border: 1px solid black;
}

.button-container {
  display: flex;
  flex-direction: column;
}

.group-1,
.group-2,
.group-3,
.group-4,
.group-5 {
  height: 100px;
  display: flex;
}

ul {
  list-style-type: none;
  display: flex;
}

li {
  margin-right: 30px;
  font-size: 2.2rem;
}

a {
  text-decoration: none;
}
