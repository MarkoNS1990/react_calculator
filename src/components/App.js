import React ,{Component} from 'react'
import '../styles/App.css';
import Display from './Display';
import ButtonPanel from './ButtonPanel';
import calculate from '../logic/calculate';

class App extends Component {
  constructor() {
    super();
    this.state = {
      total: null,
      next: null,
      operation: null,
    };
  }

  handleClick = (buttonName) => {
    const calculator = calculate(this.state, buttonName);
    this.setState({
      calculator,
    });
  }

  render() {
    const { total, next } = this.state;
    let show = total;
    if (next) {
      show = next;
    }
    if (show === null) {
      show = '0';
    }
    return (
      <>
        <Display result={show.toString()} />
        <ButtonPanel clickHandler={this.handleClick} />
      </>

    );
  }
}

export default App;
