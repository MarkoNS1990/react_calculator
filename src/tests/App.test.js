import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import App from '../components/App';
import '@testing-library/jest-dom';

it('match the snapshot', () => {
  const tree = renderer
    .create(<App />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly to Calculator page', () => {
  render(<App />);
  expect(screen.getByText('Calculator')).toBeInTheDocument();
});