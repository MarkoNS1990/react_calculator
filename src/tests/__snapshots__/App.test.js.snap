// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match the snapshot 1`] = `
Array [
  <div>
    <nav>
      <ul>
        <li>
          <a
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li>
          <a
            href="/calculator"
            onClick={[Function]}
          >
            Calculator
          </a>
        </li>
        <li>
          <a
            href="/quote"
            onClick={[Function]}
          >
            Quote of the day
          </a>
        </li>
      </ul>
    </nav>
  </div>,
  <div
    className="container"
  >
    <div>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Ipsum laboriosam unde reprehenderit labore iusto asperiores sapiente dolores cumque totam enim eius, non amet expedita, sed eveniet commodi beatae quaerat vitae!
      <p>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit. Dolorum aperiam dolores, enim in quasi quisquam? Cumque consequatur aliquam fugit quam beatae dolore ipsum eos, libero iste dolor excepturi iure ullam.
      </p>
    </div>
  </div>,
]
`;
